code/modules/paperwork/paper.dm

```diff
@@ -299,9 +297,86 @@
 		img = P
 		user.drop_item(P, src)
 		user << "<span class='notice'>You attach the photo to the piece of paper.</span>"
+	else if(P.is_hot())
+		src.burn_paper(user)
+		return //no fingerprints, paper is gone
 	add_fingerprint(user)
+	return ..()
+
+/obj/item/proc/burn_paper(mob/user)
+	var/prot = 0
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if (M_RESIST_HEAT in H.mutations)
+			prot = 1
+		else if(H.gloves)
+			var/obj/item/clothing/gloves/G = H.gloves
+			if(G.max_heat_protection_temperature)
+				prot = (G.max_heat_protection_temperature > AUTOIGNITION_PAPER)
```

Comment:
change AUTOIGNITION_PAPER here to autoignition_temperature, and set that var for the folded paper items to be AUTOIGNITION_PAPER.